Fix heating rate error

Crash-Evans opened the pull request.

Crash-Evans committed:
Intial Commit

Crash-Evans committed:
Update tado_temp_offset.yaml

Refactor Tado temperature offset blueprint

- Removed the fixed value from the `tado_temp` variable, which was previously set to 19. The reason for this specific value was unclear. Updated to dynamically fetch the current temperature from the Tado entity, enhancing flexibility and adaptability.
- Enhanced condition checking for `heating_rate` to ensure proper accumulation or comparison of values for optimal performance.
- Simplified condition logic for trigger checks to improve readability and maintainability.
- Fixed condition for `run_flag` to handle cases where `night_switch` is not provided, ensuring proper behavior.
- Revised debug logging messages for better readability.
- Ensured consistency in formatting throughout the blueprint.
- Added comments within the blueprint to explain complex logic and rationale behind decisions for easier maintenance and modification in the future.

Crash-Evans committed:
Handle error gracefully in heating_rate variable assignment

Crash-Evans committed:
Merge branch 'fix-heating-rate-error' into main

Crash-Evans committed:
Merge pull request #1 from Crash-Evans/main

Update tado_temp_offset.yaml

Clemensao commented on blueprints/automation/tado_temp_offset.yaml:
```diff
@@ -43,35 +45,30 @@ blueprint:
-    {% endfor %}
+  actual_temp: "{{ states(source_temp_sensor) | float }}"
+  run_flag: '{{ night_switch is not none and is_state(night_switch, "off") }}'
+  heating_rate: '{{ state_attr(target_tado, "heating_rate") | float }}'  # Adjust based on Tado's attributes
```
This won't work in my configuration. heating_rate is no attribute of the climate entity but a seperate sensor-entity connected to the device. By default, the tado integration creates this sensor with an entity_id ending on "heating".

I suggest this approach:
`heating_rate: '{{ expand(device_entities(device_id(target_tado))) | selectattr("entity_id","match", "^sensor.*heating$") | map(attribute="state") | first | float }}'`

Crash-Evans replied:
I'm really new to this so it's great to learn! Agree as it threw up errors for me.

Clemensao commented on blueprints/automation/tado_temp_offset.yaml:
```diff
@@ -43,35 +45,30 @@ blueprint:
-      {% endif %}
-    {% endfor %}
+  actual_temp: "{{ states(source_temp_sensor) | float }}"
+  run_flag: '{{ night_switch is not none and is_state(night_switch, "off") }}'
```
What will happen if night_switch is not set? Nighttime is an optional input! 
I admit "run_flag" is not a good variable name. It should be s.th. like day_flag

`run_flag: '{{ night_switch is none or (night_switch is not none and is_state(night_switch, "off")) }}' # flag for determining the daytime-mode`

Crash-Evans replied:
I'll have a play about and see what it the debug looks like

Clemensao commented on blueprints/automation/tado_temp_offset.yaml:
```diff
@@ -95,29 +93,38 @@ condition:
+        value_template: "{{ offset != 0 and actual_temp != 0 and heating_rate <= heating_threshold }}"
+      - condition: or
+        conditions:
+          - condition: template
```
I'd rather delete this condition (line 99 + 100)

Clemensao reviewed (changes requested):
Good work! Thanks for fixing my sloppy template :)
Please see my comments...

Cheers!

Crash-Evans commented:
I'll have a look at this over the next couple of days and see what comes back

Crash-Evans committed:
Update tado_temp_offset.yaml

Summary:
- Updated the logic for determining the heating rate to use a separate sensor entity instead of an attribute of the climate entity. This ensures compatibility with configurations where the heating rate is provided by a sensor entity ending with "heating".
- Clarified the potential issue if the `night_switch` input is not set, emphasizing its optional nature.
- Acknowledged the naming issue with the variable `run_flag` and suggested renaming it to something more appropriate like `day_flag`.
- Removed the unnecessary condition on lines 99 and 100 to enhance simplicity and readability.

Changes:
1. Updated the logic for determining the heating rate to use a separate sensor entity instead of an attribute of the climate entity. This ensures compatibility with configurations where the heating rate is provided by a sensor entity ending with "heating".

The new approach is :
heating_rate: '{{ expand(device_entities(device_id(target_tado))) | selectattr("entity_id","match", "^sensor.*heating$") | map(attribute="state") | first | float }}'

2. Clarified the role of the `night_switch` input as optional.
3. Renamed the `run_flag` variable to `day_flag` for improved clarity.
4. Removed the unnecessary condition on lines 99 and 100 for simplification and readability.

Crash-Evans committed:
Merge pull request #2 from Crash-Evans/Crash-Evans-patch-1

Update tado_temp_offset.yaml

Crash-Evans committed:
Update tado_temp_offset.yaml

Updated blueprint to prevent constant automation reloading

Description:
Removed the event trigger for 'automation_reloaded' to prevent constant reloading of the automation. Relied solely on state and time pattern triggers to activate the automation. This change ensures the automation functions as intended without unnecessary reloading.

Crash-Evans committed:
Update tado_temp_offset.yaml

Changed url

Crash-Evans committed:
Update tado_temp_offset.yaml

Crash-Evans committed:
Update tado_temp_offset.yaml

Crash-Evans committed:
Update tado_temp_offset.yaml

Changed if statement.
